Run python system_tests.py -v
  python system_tests.py -v
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.11.14/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.11.14/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.14/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.14/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.14/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.11.14/x64/lib
    CI: true
    NEXTCLOUD_DAEMON_LOG_STDOUT: 1
test_basic_file_upload (__main__.NextcloudSystemTests.test_basic_file_upload)
Test basic file upload functionality ... ok
test_conflict_resolution (__main__.NextcloudSystemTests.test_conflict_resolution)
Test filename conflict resolution ... ok
test_daemon_connection_failure_handling (__main__.NextcloudSystemTests.test_daemon_connection_failure_handling)
Test daemon behavior when Nextcloud is unavailable ... ok
test_file_deletion_after_upload (__main__.NextcloudSystemTests.test_file_deletion_after_upload)
Test that files are deleted locally after successful upload ... ok
test_file_modification_update (__main__.NextcloudSystemTests.test_file_modification_update)
Running in CI environment - assuming Nextcloud is already started
Waiting for Nextcloud to be ready...
Nextcloud is ready!
System tests using temp directory: /tmp/tmpbkemki2w
Daemon returncode: -15
Daemon stdout: 2025-10-18 18:16:28,668 - nextcloud_upload_daemon[3376]: INFO - Starting Nextcloud Upload Daemon
2025-10-18 18:16:29,128 - nextcloud_upload_daemon[3376]: INFO - Successfully connected to Nextcloud server
2025-10-18 18:16:29,128 - nextcloud_upload_daemon[3376]: INFO - Using upload delay: 1 seconds
2025-10-18 18:16:29,128 - nextcloud_upload_daemon[3376]: INFO - Using delete delay: 3 seconds
2025-10-18 18:16:29,128 - nextcloud_upload_daemon[3376]: INFO - Monitoring directory: /tmp/tmpbkemki2w/upload/test_basic_file_upload -> /test_test_basic_file_upload
2025-10-18 18:16:29,129 - nextcloud_upload_daemon[3376]: INFO - Nextcloud Upload Daemon started successfully
Nextcloud Upload Daemon started. Press Ctrl+C to stop.
2025-10-18 18:16:32,696 - nextcloud_upload_daemon[3376]: INFO - Creating directory /test_test_basic_file_upload
DEBUG: Creating directory /test_test_basic_file_upload
2025-10-18 18:16:32,978 - nextcloud_upload_daemon[3376]: INFO - Successfully created directory /test_test_basic_file_upload
DEBUG: Successfully created directory /test_test_basic_file_upload
2025-10-18 18:16:33,421 - nextcloud_upload_daemon[3376]: INFO - Successfully uploaded /tmp/tmpbkemki2w/upload/test_basic_file_upload/test_upload.txt as test_upload.txt to /test_test_basic_file_upload
DEBUG: Successfully uploaded /tmp/tmpbkemki2w/upload/test_basic_file_upload/test_upload.txt as test_upload.txt to /test_test_basic_file_upload
2025-10-18 18:16:36,422 - nextcloud_upload_daemon[3376]: INFO - Deleted local file /tmp/tmpbkemki2w/upload/test_basic_file_upload/test_upload.txt after successful upload

Daemon returncode: -15
Daemon stdout: 2025-10-18 18:16:44,278 - nextcloud_upload_daemon[3408]: INFO - Starting Nextcloud Upload Daemon
2025-10-18 18:16:44,460 - nextcloud_upload_daemon[3408]: INFO - Successfully connected to Nextcloud server
2025-10-18 18:16:44,460 - nextcloud_upload_daemon[3408]: INFO - Using upload delay: 1 seconds
2025-10-18 18:16:44,460 - nextcloud_upload_daemon[3408]: INFO - Using delete delay: 3 seconds
2025-10-18 18:16:44,460 - nextcloud_upload_daemon[3408]: INFO - Monitoring directory: /tmp/tmpbkemki2w/upload/test_conflict_resolution -> /test_test_conflict_resolution
2025-10-18 18:16:44,461 - nextcloud_upload_daemon[3408]: INFO - Nextcloud Upload Daemon started successfully
Nextcloud Upload Daemon started. Press Ctrl+C to stop.
2025-10-18 18:16:48,356 - nextcloud_upload_daemon[3408]: INFO - Creating directory /test_test_conflict_resolution
DEBUG: Creating directory /test_test_conflict_resolution
2025-10-18 18:16:48,561 - nextcloud_upload_daemon[3408]: INFO - Successfully created directory /test_test_conflict_resolution
DEBUG: Successfully created directory /test_test_conflict_resolution
2025-10-18 18:16:48,968 - nextcloud_upload_daemon[3408]: INFO - Successfully uploaded /tmp/tmpbkemki2w/upload/test_conflict_resolution/common_name.txt as common_name.txt to /test_test_conflict_resolution
DEBUG: Successfully uploaded /tmp/tmpbkemki2w/upload/test_conflict_resolution/common_name.txt as common_name.txt to /test_test_conflict_resolution
2025-10-18 18:16:51,969 - nextcloud_upload_daemon[3408]: INFO - Deleted local file /tmp/tmpbkemki2w/upload/test_conflict_resolution/common_name.txt after successful upload

Daemon returncode: -15
Daemon stdout: 2025-10-18 18:16:57,461 - nextcloud_upload_daemon[3476]: INFO - Starting Nextcloud Upload Daemon
2025-10-18 18:16:57,640 - nextcloud_upload_daemon[3476]: INFO - Successfully connected to Nextcloud server
2025-10-18 18:16:57,640 - nextcloud_upload_daemon[3476]: INFO - Using upload delay: 1 seconds
2025-10-18 18:16:57,640 - nextcloud_upload_daemon[3476]: INFO - Using delete delay: 3 seconds
2025-10-18 18:16:57,640 - nextcloud_upload_daemon[3476]: INFO - Monitoring directory: /tmp/tmpbkemki2w/upload/test_conflict_resolution -> /test_test_conflict_resolution
2025-10-18 18:16:57,641 - nextcloud_upload_daemon[3476]: INFO - Nextcloud Upload Daemon started successfully
Nextcloud Upload Daemon started. Press Ctrl+C to stop.
2025-10-18 18:17:02,116 - nextcloud_upload_daemon[3476]: INFO - Successfully uploaded /tmp/tmpbkemki2w/upload/test_conflict_resolution/common_name.txt as Copy_2-common_name.txt to /test_test_conflict_resolution
DEBUG: Successfully uploaded /tmp/tmpbkemki2w/upload/test_conflict_resolution/common_name.txt as Copy_2-common_name.txt to /test_test_conflict_resolution
2025-10-18 18:17:05,117 - nextcloud_upload_daemon[3476]: INFO - Deleted local file /tmp/tmpbkemki2w/upload/test_conflict_resolution/common_name.txt after successful upload

Daemon returncode: -15
Daemon stdout: 2025-10-18 18:17:15,449 - nextcloud_upload_daemon[3551]: INFO - Starting Nextcloud Upload Daemon
2025-10-18 18:17:15,627 - nextcloud_upload_daemon[3551]: INFO - Successfully connected to Nextcloud server
2025-10-18 18:17:15,627 - nextcloud_upload_daemon[3551]: INFO - Using upload delay: 1 seconds
2025-10-18 18:17:15,627 - nextcloud_upload_daemon[3551]: INFO - Using delete delay: 3 seconds
2025-10-18 18:17:15,628 - nextcloud_upload_daemon[3551]: INFO - Monitoring directory: /tmp/tmpbkemki2w/upload/test_file_deletion_after_upload -> /test_test_file_deletion_after_upload
2025-10-18 18:17:15,628 - nextcloud_upload_daemon[3551]: INFO - Nextcloud Upload Daemon started successfully
Nextcloud Upload Daemon started. Press Ctrl+C to stop.
2025-10-18 18:17:19,521 - nextcloud_upload_daemon[3551]: INFO - Creating directory /test_test_file_deletion_after_upload
DEBUG: Creating directory /test_test_file_deletion_after_upload
2025-10-18 18:17:19,723 - nextcloud_upload_daemon[3551]: INFO - Successfully created directory /test_test_file_deletion_after_upload
DEBUG: Successfully created directory /test_test_file_deletion_after_upload
2025-10-18 18:17:20,121 - nextcloud_upload_daemon[3551]: INFO - Successfully uploaded /tmp/tmpbkemki2w/upload/test_file_deletion_after_upload/test_delete.txt as test_delete.txt to /test_test_file_deletion_after_upload
DEBUG: Successfully uploaded /tmp/tmpbkemki2w/upload/test_file_deletion_after_upload/test_delete.txt as test_delete.txt to /test_test_file_deletion_after_upload
2025-10-18 18:17:23,122 - nextcloud_upload_daemon[3551]: INFO - Deleted local file /tmp/tmpbkemki2w/upload/test_file_deletion_after_upload/test_delete.txt after successful upload

Daemon returncode: -15
Daemon stdout: 2025-10-18 18:17:30,885 - nextcloud_upload_daemon[3583]: INFO - Starting Nextcloud Upload Daemon
2025-10-18 18:17:31,065 - nextcloud_upload_daemon[3583]: INFO - Successfully connected to Nextcloud server
2025-10-18 18:17:31,065 - nextcloud_upload_daemon[3583]: INFO - Using upload delay: 1 seconds
2025-10-18 18:17:31,065 - nextcloud_upload_daemon[3583]: INFO - Using delete delay: 3 seconds
2025-10-18 18:17:31,065 - nextcloud_upload_daemon[3583]: INFO - Monitoring directory: /tmp/tmpbkemki2w/upload/test_file_modification_update -> /test_test_file_modification_update
2025-10-18 18:17:31,066 - nextcloud_upload_daemon[3583]: INFO - Nextcloud Upload Daemon started successfully
Nextcloud Upload Daemon started. Press Ctrl+C to stop.
2025-10-18 18:17:34,964 - nextcloud_upload_daemon[3583]: INFO - Creating directory /test_test_file_modification_update
DEBUG: Creating directory /test_test_file_modification_update
2025-10-18 18:17:35,177 - nextcloud_upload_daemon[3583]: INFO - Successfully created directory /test_test_file_modification_update
DEBUG: Successfully created directory /test_test_file_modification_update
2025-10-18 18:17:35,585 - nextcloud_upload_daemon[3583]: INFO - Successfully uploaded /tmp/tmpbkemki2w/upload/test_file_modification_update/test_modify.txt as test_modify.txt to /test_test_file_modification_update
DEBUG: Successfully uploaded /tmp/tmpbkemki2w/upload/test_file_modification_update/test_modify.txt as test_modify.txt to /test_test_file_modification_update
2025-10-18 18:17:38,586 - nextcloud_upload_daemon[3583]: INFO - Deleted local file /tmp/tmpbkemki2w/upload/test_file_modification_update/test_modify.txt after successful upload
2025-10-18 18:17:40,718 - nextcloud_upload_daemon[3583]: INFO - Successfully uploaded /tmp/tmpbkemki2w/upload/test_file_modification_update/test_modify.txt as Copy_2-test_modify.txt to /test_test_file_modification_update
DEBUG: Successfully uploaded /tmp/tmpbkemki2w/upload/test_file_modification_update/test_modify.txt as Copy_2-test_modify.txt to /test_test_file_modification_update
2025-10-18 18:17:43,719 - nextcloud_upload_daemon[3583]: INFO - Deleted local file /tmp/tmpbkemki2w/upload/test_file_modification_update/test_modify.txt after successful upload
Test that modified files are updated in Nextcloud ... FAIL
test_multiple_files_upload (__main__.NextcloudSystemTests.test_multiple_files_upload)
Test uploading multiple files simultaneously ... ok
test_webdav_authentication (__main__.NextcloudSystemTests.test_webdav_authentication)
Test WebDAV authentication with Nextcloud ... ok

======================================================================
FAIL: test_file_modification_update (__main__.NextcloudSystemTests.test_file_modification_update)
Test that modified files are updated in Nextcloud
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/nextcloud_upload_daemon/nextcloud_upload_daemon/system_tests.py", line 337, in test_file_modification_update
    self.assertEqual(uploaded_content, modified_content, "Updated content should be reflected in Nextcloud")
AssertionError: 'Original content' != 'Modified content - updated!'
- Original content
+ Modified content - updated!
 : Updated content should be reflected in Nextcloud

----------------------------------------------------------------------
Ran 7 tests in 109.034s

FAILED (failures=1)

Daemon returncode: -15
Daemon stdout: 2025-10-18 18:17:47,512 - nextcloud_upload_daemon[3658]: INFO - Starting Nextcloud Upload Daemon
2025-10-18 18:17:47,694 - nextcloud_upload_daemon[3658]: INFO - Successfully connected to Nextcloud server
2025-10-18 18:17:47,694 - nextcloud_upload_daemon[3658]: INFO - Using upload delay: 1 seconds
2025-10-18 18:17:47,694 - nextcloud_upload_daemon[3658]: INFO - Using delete delay: 3 seconds
2025-10-18 18:17:47,694 - nextcloud_upload_daemon[3658]: INFO - Monitoring directory: /tmp/tmpbkemki2w/upload/test_multiple_files_upload -> /test_test_multiple_files_upload
2025-10-18 18:17:47,695 - nextcloud_upload_daemon[3658]: INFO - Nextcloud Upload Daemon started successfully
Nextcloud Upload Daemon started. Press Ctrl+C to stop.
2025-10-18 18:17:51,585 - nextcloud_upload_daemon[3658]: INFO - Creating directory /test_test_multiple_files_upload
DEBUG: Creating directory /test_test_multiple_files_upload
2025-10-18 18:17:51,786 - nextcloud_upload_daemon[3658]: INFO - Successfully created directory /test_test_multiple_files_upload
DEBUG: Successfully created directory /test_test_multiple_files_upload
2025-10-18 18:17:52,186 - nextcloud_upload_daemon[3658]: INFO - Successfully uploaded /tmp/tmpbkemki2w/upload/test_multiple_files_upload/file1.txt as file1.txt to /test_test_multiple_files_upload
DEBUG: Successfully uploaded /tmp/tmpbkemki2w/upload/test_multiple_files_upload/file1.txt as file1.txt to /test_test_multiple_files_upload
2025-10-18 18:17:52,808 - nextcloud_upload_daemon[3658]: INFO - Successfully uploaded /tmp/tmpbkemki2w/upload/test_multiple_files_upload/file2.txt as file2.txt to /test_test_multiple_files_upload
DEBUG: Successfully uploaded /tmp/tmpbkemki2w/upload/test_multiple_files_upload/file2.txt as file2.txt to /test_test_multiple_files_upload
2025-10-18 18:17:53,395 - nextcloud_upload_daemon[3658]: INFO - Successfully uploaded /tmp/tmpbkemki2w/upload/test_multiple_files_upload/file3.txt as file3.txt to /test_test_multiple_files_upload
DEBUG: Successfully uploaded /tmp/tmpbkemki2w/upload/test_multiple_files_upload/file3.txt as file3.txt to /test_test_multiple_files_upload
2025-10-18 18:17:53,971 - nextcloud_upload_daemon[3658]: INFO - Successfully uploaded /tmp/tmpbkemki2w/upload/test_multiple_files_upload/subdoc.md as subdoc.md to /test_test_multiple_files_upload
DEBUG: Successfully uploaded /tmp/tmpbkemki2w/upload/test_multiple_files_upload/subdoc.md as subdoc.md to /test_test_multiple_files_upload
2025-10-18 18:17:55,187 - nextcloud_upload_daemon[3658]: INFO - Deleted local file /tmp/tmpbkemki2w/upload/test_multiple_files_upload/file1.txt after successful upload
2025-10-18 18:17:55,808 - nextcloud_upload_daemon[3658]: INFO - Deleted local file /tmp/tmpbkemki2w/upload/test_multiple_files_upload/file2.txt after successful upload
2025-10-18 18:17:56,396 - nextcloud_upload_daemon[3658]: INFO - Deleted local file /tmp/tmpbkemki2w/upload/test_multiple_files_upload/file3.txt after successful upload
2025-10-18 18:17:56,972 - nextcloud_upload_daemon[3658]: INFO - Deleted local file /tmp/tmpbkemki2w/upload/test_multiple_files_upload/subdoc.md after successful upload

Error: Process completed with exit code 1.
